packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -120,48 +294,55 @@ typedef OnInvokeCallback = void Function(FocusNode node, Intent tag);
 ///    and allows redefining of actions for its descendants.
 ///  * [ActionDispatcher], a class that takes an [Action] and invokes it using a
 ///    [FocusNode] for context.
-class CallbackAction extends Action {
-  /// A const constructor for an [Action].
+class CallbackAction<T extends Intent> extends Action<T> {
+  /// A constructor for a [CallbackAction].
   ///
   /// The `intentKey` and [onInvoke] parameters must not be null.
   /// The [onInvoke] parameter is required.
-  const CallbackAction(LocalKey intentKey, {@required this.onInvoke})
-      : assert(onInvoke != null),
-        super(intentKey);
+  CallbackAction({@required this.onInvoke}) : assert(onInvoke != null);
 
   /// The callback to be called when invoked.
   ///
   /// Must not be null.
   @protected
-  final OnInvokeCallback onInvoke;
+  final OnInvokeCallback<T> onInvoke;
 
   @override
-  void invoke(FocusNode node, Intent intent) => onInvoke.call(node, intent);
+  Object invoke(covariant T intent) => onInvoke.call(intent);
```

Comment:
nit: `onInvoke(intent)` seems a little clearer to me then the indirection via call.